kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -108,7 +107,10 @@ internal class KotlinJsonAdapter<T>(
     }
 
     // Call the constructor using a Map so that absent optionals get defaults.
-    val result = constructor.callBy(IndexedParameterMap(constructor.parameters, values))
+    val result = if (values.isFullInitialized())
+      SpreadWrapper.call(constructor, values.parameterValues)
+    else
+      constructor.callBy(values)
```

Comment:
Please match our style

```suggestion
    val result = if (values.isFullInitialized()) {
      SpreadWrapper.call(constructor, values.parameterValues)
    } else {
      constructor.callBy(values)
    }
```

Reply:
I have fixed it.

https://github.com/square/moshi/pull/1286/files#diff-129f467727ea6286b698c1c3be4e4844742f064fc0aeacae5af5dc338d7fe3e4R113-R117